src/Jupyter/Visualization/DisplayableExceptionEncoders.cs

```diff
@@ -2,27 +2,38 @@
 // Licensed under the MIT License.
 #nullable enable
 
-using System;
-using System.Collections;
-using System.Collections.Generic;
-using System.Collections.Immutable;
+using Microsoft.Quantum.Simulation.Common;
+using Microsoft.Quantum.Simulation.Core;
 using System.Data;
-using System.Linq;
-using System.Net;
 using System.Text;
 using System.Text.RegularExpressions;
-using Microsoft.Jupyter.Core;
-using Microsoft.Quantum.QsCompiler.SyntaxTree;
-using Microsoft.Quantum.Simulation.Common;
-using Microsoft.Quantum.Simulation.Core;
-using Microsoft.Quantum.Simulation.Simulators;
 
 namespace Microsoft.Quantum.IQSharp
 {
-    internal record struct DisplayableException(Exception Exception, IEnumerable<StackFrame>? StackTrace)
+    internal record struct DisplayableStackFrame(string Callable, string SourceFile, string BestSourceLocation, int LineNumber)
     {
+    }
+
+    internal record struct DisplayableException(string? ExceptionType, string ExceptionMessage, IEnumerable<DisplayableStackFrame> StackTrace)
+    {
+        public static DisplayableException Create(Exception exception, IEnumerable<StackFrame> stackTrace)
```

Comment:
The problematic type is [`StackFrame`](https://github.com/microsoft/qsharp-runtime/blob/09403e3d306567f2cff237881dbb064829b496a5/src/Simulation/Simulators/CommonNativeSimulator/StackTrace.cs#L17), which has a reference to `Callable`, which can have references to `Assembly` objects, etc., etc., which ultimately fan out into hundreds of types to be serialized. We shouldn't have this type in a class that's intended to be serialized.